File: backend/app/schemas/lenders.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel
from pydantic import EmailStr
from pydantic import Field


class LenderCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    contact_name: str | None = Field(default=None, max_length=255)
    contact_email: EmailStr | None = None
    contact_phone: str | None = Field(default=None, max_length=50)
    notes: str | None = None


class LenderUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    contact_name: str | None = Field(default=None, max_length=255)
    contact_email: EmailStr | None = None
    contact_phone: str | None = Field(default=None, max_length=50)
    notes: str | None = None


class LenderListItem(BaseModel):
    id: UUID
    name: str
    contact_name: str | None = None
    contact_email: str | None = None
    contact_phone: str | None = None
    notes: str | None = None
    active_facility_count: int
    created_at: datetime
    updated_at: datetime


class LenderFacilityLink(BaseModel):
    facility_id: UUID
    property_id: UUID | None = None
    property_address: str | None = None
    lender_type: str
    status: str
    total_facility: str | None = None
    opening_balance: str | None = None


class LenderDetail(LenderListItem):
    facilities: list[LenderFacilityLink]

File: frontend/src/types/lenders.ts
export interface LenderListItem {
  id: string;
  name: string;
  contact_name?: string | null;
  contact_email?: string | null;
  contact_phone?: string | null;
  notes?: string | null;
  active_facility_count: number;
  created_at: string;
  updated_at: string;
}

export interface LenderFacilityLink {
  facility_id: string;
  property_id?: string | null;
  property_address?: string | null;
  lender_type: string;
  status: string;
  total_facility?: string | null;
  opening_balance?: string | null;
}

export interface LenderDetail extends LenderListItem {
  facilities: LenderFacilityLink[];
}

export interface LenderPayload {
  name: string;
  contact_name?: string | null;
  contact_email?: string | null;
  contact_phone?: string | null;
  notes?: string | null;
}
